lib/jsonApi.js

```diff
@@ -107,7 +107,14 @@ jsonApi.onUncaughtException = function(errHandler) {
 
 jsonApi.start = function() {
   routes.register();
-  router.listen(jsonApi._apiConfig.port);
+  var serverFactory = function(app) {
+      if (jsonApi._apiConfig.protocol === "https") {
+        return require("https").createServer(jsonApi._apiConfig.tls || {}, app);
+      } else {
+        return require("http").createServer(app);
```

Comment:
I find myself wondering if this could be easier to follow if we moved this `serverFactory` into `router.js` to keep all of the HTTP(s) routing code in one place - everything you need is in scope, this can simply be lifted from here into the `router.listen` function. I'm happy to wait and let you make the change, or I'm also happy to merge this as-is and I'll come back to it later?